### PR TITLE
Docs: Updates and improvements

### LINKED ISSUE #240: Update documentation
- [ ] CONTRIBUTING: Instruction for building backend is unclear. “build:backend” is listed as the build command, but should be “mage”
- [ ] CONTRIBUTING: Update team name here https://github.com/grafana/timestream-datasource/blob/main/CONTRIBUTING.md#re-generating-json-api-response where it says “ping the @grafana/aws-plugins team”
- [ ] CONTRIBUTING: https://github.com/grafana/timestream-datasource/blob/main/CONTRIBUTING.md#pre-requisites link to release.yaml is a 404
- [ ] README: https://github.com/grafana/timestream-datasource/blob/main/src/README.md#query-editor image link to “query-editor” is broken
- [ ] README: https://github.com/grafana/timestream-datasource/blob/main/src/README.md#alerting link to “Alerting documentation” is a 404
- [ ] README: doesn’t have documentation for provisioning a data source. See the [Athena data source README](https://github.com/grafana/athena-datasource/blob/91394b62e7653e8d9a8db6e99f015a68f6bb830b/README.md?plain=1#L186-L267) for an example of what to include

### PR DESCRIPTION
**Pull request by nmarrs**
Update and improve Timestream data source docs

Fixes #240

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -66,16 +66,6 @@ Golden files check that data frames are being generated correctly based on the T
 
 ## Build a release for the Timestream data source plugin
 
-### Pre-requisites
```
Comment (nmarrs) [inline CONTRIBUTING.md:69]: see https://github.com/grafana/timestream-datasource/issues/240#issuecomment-2316414689 for context on this

File: src/README.md [hunk 2]
```diff
@@ -35,7 +35,7 @@ Here is a policy example:
 
 The query editor accepts timestream syntax in addition to the macros listed above and any dashboard template variables.
 
-![query-editor](https://storage.googleapis.com/plugins-ci/plugins/timestream/timestream-query.png)
+![query-editor](./img/query_editor.png)
```
Comment (nmarrs) [inline src/README.md:38]: 🌳 
Screenshot 2024-08-28 at 6 10 53 PM

File: src/README.md [hunk 3]
```diff
@@ -70,9 +70,7 @@ Read more about variable formatting options in the [Variables](https://grafana.c
 
 ### Alerting
 
-See the [Alerting](https://grafana.com/docs/grafana/latest/alerting/alerts-overview/) documentation for more on Grafana alerts.
-
-[Unified Alerting](https://grafana.com/docs/grafana/latest/alerting/unified-alerting/) queries should contain a time series field. Queries without this field will return an error: "input data must be a wide series but got type long". To return time series, you can use the [`CREATE_TIME_SERIES` function](https://docs.aws.amazon.com/timestream/latest/developerguide/timeseries-specific-constructs.views.html). For example:
+[Alerting](https://grafana.com/docs/grafana/latest/alerting/) queries should contain a time series field. Queries without this field will return an error: "input data must be a wide series but got type long". To return time series, you can use the [`CREATE_TIME_SERIES` function](https://docs.aws.amazon.com/timestream/latest/developerguide/timeseries-specific-constructs.views.html). For example:
```
Comment (nmarrs) [inline src/README.md:73]: honestly not sure if this is still a constraint but I do know `unified alerting` is just alerting now 😬 [see this video for some fun historical context on history of alerting at Grafana](https://www.youtube.com/watch?v=7z5FwdfL7v4)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -66,16 +66,6 @@ Golden files check that data frames are being generated correctly based on the T
 
 ## Build a release for the Timestream data source plugin
```
Comment (idastambuk) [inline CONTRIBUTING.md:67]: Maybe we can remove this heading too and replace it with just the ##Release one

**Review by idastambuk — APPROVED** [review]

**Pull request merged**